scripts/task/ExportDeliveryClassXmlResults.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace oat\taoOutcomeUi\scripts\task;
+
+use common_report_Report as Report;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\taoDelivery\model\execution\ServiceProxy;
+use oat\taoOutcomeUi\model\ResultsService;
+use oat\taoResultServer\models\classes\QtiResultsService;
+
+/**
+ * Run example:
+ *
+ * http://www.tao.lu/Ontologies/TAODelivery.rdf\#AssembledDelivery
+ * sudo -u www-data php index.php 'oat\taoOutcomeUi\scripts\task\ExportDeliveryClassXmlResults' -u {deliveryClassUri}
+ *
+ */
+class ExportDeliveryClassXmlResults extends ScriptAction
+{
+    use OntologyAwareTrait;
+
+    /**
+     * @var Report
+     */
+    private $report;
+
+    protected function provideOptions()
+    {
+        return [
+            'deliveryClassUri' => [
+                'prefix' => 'u',
+                'longPrefix' => 'uri',
+                'required' => true,
+                'description' => 'Delivery Class URI.'
+            ],
+            'zipFilenamePrefix' => [
+                'prefix' => 'p',
+                'longPrefix' => 'filename_prefix',
+                'required' => false,
+                'defaultValue' => 'xml_results',
+                'description' => 'Prefix for ZIP file name'
+            ],
+        ];
+    }
+
+    protected function provideDescription()
+    {
+        return 'The script will generate a zip with xml results';
+    }
+
+    protected function provideUsage()
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+        ];
+    }
+
+    protected function run()
+    {
+        $this->report = Report::createInfo('Export started');
+
+        $class = $this->getOption('deliveryClassUri');
+        $zipFilenamePrefix = $this->getOption('zipFilenamePrefix');
+        $deliveryClass = $this->getClass($class);
+        $resultCount = 0;
+
+        $fileName = \tao_helpers_Display::textCleaner($zipFilenamePrefix) .'-'. time() . '.zip';
+        $path = \tao_helpers_File::concat([\tao_helpers_Export::getExportPath(), $fileName]);
+
+        if (!\tao_helpers_File::securityCheck($path, true)) {
+            throw new \Exception('Unauthorized file name');
```

Comment:
This exception is not in a try block, so it will not be caught in a catch block below in case of throwing. Is it expected behavior?

Reply:
fixed